### ISSUE TITLE
Question: index on property (across all types)

### ISSUE THREAD

**Issue by justin2004** [ArcadeData/arcadedb#1861]
Once the database starts getting populated with nodes and edges gremlin queries like this start executing slowly:
```
g.V().has("uri","uri://a10").as("s").V().has("uri","uri://a10/description").as("o").addE("described_by").from("s").to("o").next()
```

However, if I do this:
```
create index on `typeHere` (`uri`) unique
create index on `anotherTypeHere` (`uri`) unique
```

Then run queries like this then they execute quickly:
```
g.V().hasLabel('typeHere').has("uri","uri://a10").as("s").V().hasLabel('anotherTypeHere').has("uri","uri://a10/description").as("o").addE("described_by").from("s").to("o").next()
```

The trouble is, we don't always know what kind/type/label of vertex we are looking for. Mostly we only know its `uri` property value and we want to get the vertex with that particular `uri` value.

Is it possible to create an index on the property `uri` across all types so that gremlin queries like this:
```
g.V().has("uri","uri://a10").as("s").V().has("uri","uri://a10/description").as("o").addE("described_by").from("s").to("o").next()
```
always use an index for looking up the `uri` value?

In other words, is it possible to create a single index on a property (that all nodes (of all types) will have) that gremlin queries will use?

### ArcadeDB Version:

v.23.10.1-SNAPSHOT 
Runs on Linux 5.15.0-125-generic - OpenJDK 64-Bit Server VM 11.0.24 (Temurin-11.0.24+8)

**Comment by lvca** [comment]
Yes, create a base vertex type, like 'Node' with the uri indexed property and then let all the vertex types extend from that base type.

**Comment by lvca** [comment]
If you have tons of types can actually slow down things. I suggest also to create vertex type with 1 bucket only if you have many vertex types. This will speed up such lookup.

**Comment by lvca** [comment]
Closing this issue unless there is more feedback from the author.
